
  0%|                                                                                                                                             | 0/1639 [01:14<?, ?it/s]
Traceback (most recent call last):
  File "/local/a/jgammell/dlsca_bag_of_tricks/src/main.py", line 111, in run_wandb_trial_
    trainer.train_model(settings['total_epochs'], results_save_dir=results_save_dir, model_save_dir=model_save_dir)
  File "/local/a/jgammell/dlsca_bag_of_tricks/src/train/weird_gan.py", line 456, in train_model
    train_erv = self.train_epoch()
  File "/local/a/jgammell/dlsca_bag_of_tricks/src/train/weird_gan.py", line 387, in train_epoch
    step_rv = step_fn(batch, val_batch=val_batch, train_gen=train_gen, train_disc=train_disc, **kwargs)
  File "/local/a/jgammell/dlsca_bag_of_tricks/src/train/weird_gan.py", line 273, in train_step
    perturbed_train_traces.requires_grad = True
UnboundLocalError: local variable 'perturbed_train_traces' referenced before assignment